protocol/statetracker/provider_state_tracker.go

```diff
@@ -44,17 +44,19 @@ func (pst *ProviderStateTracker) RegisterForEpochUpdates(ctx context.Context, ep
 	epochUpdater.RegisterEpochUpdatable(ctx, epochUpdatable)
 }
 
-func (pst *ProviderStateTracker) RegisterChainParserForSpecUpdates(ctx context.Context, chainParser chainlib.ChainParser, chainID string) error {
-	spec, err := pst.stateQuery.GetSpec(ctx, chainID)
-	if err != nil {
-		return err
+func (pst *ProviderStateTracker) RegisterForSpecUpdates(ctx context.Context, chainParser chainlib.ChainParser, chainId string, apiInterface string) error {
+	// register for spec updates when a spec has been modified
+	specUpdater := NewSpecUpdater(chainId, chainParser, pst.stateQuery, pst.eventTracker, apiInterface)
```

Comment:
after consolidating all api interfaces to one specUpdater remove the apiInterface